// cesium/Build/CesiumUnminified/Workers/upsampleVerticesFromCesium3DTilesTerrain.js
/**
 * @license
 * Cesium - https://github.com/CesiumGS/cesium
 * Version 1.142.0
 *
 * Copyright 2011-2022 Cesium Contributors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Columbus View (Pat. Pend.)
 *
 * Portions licensed separately.
 * See https://github.com/CesiumGS/cesium/blob/main/LICENSE.md for full licensing details.
 */

import {
  Cesium3DTilesTerrainGeometryProcessor_default
} from "./chunk-424RNO3D.js";
import "./chunk-7WM2PX4V.js";
import "./chunk-VWV2IJRB.js";
import {
  createTaskProcessorWorker_default
} from "./chunk-4S7IE3LB.js";
import "./chunk-4MGMYHOO.js";
import "./chunk-EPYXRCDB.js";
import "./chunk-7J2HKPRH.js";
import "./chunk-SMQZZOW6.js";
import "./chunk-DDWSZVMY.js";
import "./chunk-3VFEWOBI.js";
import "./chunk-E7VQ76KE.js";
import "./chunk-S4MCOY6I.js";
import "./chunk-XZMBNNDX.js";
import "./chunk-C6J67V5A.js";
import "./chunk-MYP774IS.js";
import "./chunk-Q3PY526V.js";
import "./chunk-P3CK7MCY.js";
import "./chunk-TKKIREQ6.js";
import "./chunk-Y5BF6AFU.js";
import "./chunk-V7VQKN6N.js";
import "./chunk-MPZHGZU6.js";
import "./chunk-J6BWOHUF.js";
import "./chunk-AYKR4VBR.js";
import "./chunk-ZP7JMQV4.js";

// packages/engine/Source/Workers/upsampleVerticesFromCesium3DTilesTerrain.js
function upsampleVerticesFromCesium3DTilesTerrain(options, transferableObjects) {
  const mesh = Cesium3DTilesTerrainGeometryProcessor_default.upsampleMesh(options);
  const verticesBuffer = mesh.vertices.buffer;
  const indicesBuffer = mesh.indices.buffer;
  const westIndicesBuffer = mesh.westIndicesSouthToNorth.buffer;
  const southIndicesBuffer = mesh.southIndicesEastToWest.buffer;
  const eastIndicesBuffer = mesh.eastIndicesNorthToSouth.buffer;
  const northIndicesBuffer = mesh.northIndicesWestToEast.buffer;
  transferableObjects.push(
    verticesBuffer,
    indicesBuffer,
    westIndicesBuffer,
    southIndicesBuffer,
    eastIndicesBuffer,
    northIndicesBuffer
  );
  const result = {
    verticesBuffer,
    indicesBuffer,
    vertexCountWithoutSkirts: mesh.vertexCountWithoutSkirts,
    indexCountWithoutSkirts: mesh.indexCountWithoutSkirts,
    encoding: mesh.encoding,
    westIndicesBuffer,
    southIndicesBuffer,
    eastIndicesBuffer,
    northIndicesBuffer,
    minimumHeight: mesh.minimumHeight,
    maximumHeight: mesh.maximumHeight,
    boundingSphere: mesh.boundingSphere3D,
    orientedBoundingBox: mesh.orientedBoundingBox,
    horizonOcclusionPoint: mesh.horizonOcclusionPoint
  };
  return result;
}
var upsampleVerticesFromCesium3DTilesTerrain_default = createTaskProcessorWorker_default(
  upsampleVerticesFromCesium3DTilesTerrain
);
export {
  upsampleVerticesFromCesium3DTilesTerrain_default as default
};
